### PR TITLE
feat: support chunking with overlapping in ES ingest pipeline

### PR DESCRIPTION
**Pull request by zach-shu**
This PR adds a script as an option for chunking with overlapping when creating an ingest pipeline in Elasticsearch.

Signed off by: zhongzheng.shu@ibm.com

### CODE REVIEW

**Review by rfazeli — APPROVED** [review]
Thanks for putting this together, Zach! All looks good to me. The overlap scripts for both pdf ingestion and web crawler work fine for me.

File: integrations/extensions/docs/elasticsearch-install-and-setup/how_to_use_web_crawler_in_elasticsearch.md [hunk 1]
```diff
@@ -286,19 +286,69 @@ Now you can build a custom ingest pipeline for your web crawler index on Kibana,
...
+    }
     ```
+    This script splits the `body_content` into sentences using regex and combines them into `passages`. The maximum number of characters in each paggase is controlled by the `model_limit` parameter. There is a overlapping between two adjacent passages, and it is controled by the `overlap_percentage` parameter. So, `model_limit` and `overlap_percentage` need to be configured in the `Parameters` field, for example, 
+    ```json
+    {
+      "model_limit": 2048,
+      "overlap_percentage": 0.25
+    }
+    ```
+    NOTE: Chunking with 512 tokens and 25% overlapping has been shown to perform better than other strategies, so the above `Parameters` configuration is recommeded for your chunking processor.
```
Comment (jwm4) [inline integrations/extensions/docs/elasticsearch-install-and-setup/how_to_use_web_crawler_in_elasticsearch.md:350]: Do you have a source you can cite for "Chunking with 512 tokens and 25% overlapping has been shown to perform better than other strategies"?  If not, we can just leave it, but it seems like the kind of claim I would usually want to link to a source for.

**Review by jwm4 — CHANGES_REQUESTED** [review]

File: integrations/extensions/docs/elasticsearch-install-and-setup/how_to_use_web_crawler_in_elasticsearch.md (hunk 1 above)
Comment (zach-shu) [reply integrations/extensions/docs/elasticsearch-install-and-setup/how_to_use_web_crawler_in_elasticsearch.md:350]: Here is a source. https://techcommunity.microsoft.com/t5/ai-azure-ai-services-blog/azure-ai-search-outperforming-vector-search-with-hybrid/ba-p/3929167.

We don't know how the 25% overlapping was implemented in their experiments. Our implementation here is not exactly 512 tokens with 25% overlapping, but more of an approximation. Do you think we should include this source here?

Comment (jwm4) [reply integrations/extensions/docs/elasticsearch-install-and-setup/how_to_use_web_crawler_in_elasticsearch.md]: I think it is a good source that makes the point you are trying to make well.  I agree the details are a little different, but this at least gives people some sense of why we think around 25% or so is good.

**Review by jwm4 — APPROVED** [review]
The content looks good to me.  Given that @rfazeli has confirmed that everything is working, I think this is ready to merge now.

**Pull request merged**